### ISSUE TITLE
How to deploy standard environment EAR with this plugin?

### ISSUE THREAD

**Issue by swftvsn** [GoogleCloudPlatform/app-gradle-plugin#86]
Trying to deploy EAR that has three wars inside, default and two additional services. The ear builds fine and has the appengine-application.xml in the root of the ear. 

Is there any documentation how this should work?

**Comment by patflynn** [comment]
Hi @swftvsn,

The EAR file format is not supported by the new Cloud SDK based Maven and Gradle toolchain.

It is not difficult though to deploy multi-module applications using the new tools. You can simply add your other modules staged app.yaml to your default modules `deployables` config. @loosebazooka correct me if I'm wrong. 

FYI Using the approach I just suggested means you no longer need an EAR module, an appengine-application.xml, or to do any special packaging prior to deployment (apart from the build and staging of the individual web modules you want to deploy).

We are in the process of updating our documentation to make the new flow clear for users of the EAR based multi-module flows.

@loosebazooka Do we have an open bug for the docs update needed?

**Comment by loosebazooka** [comment]
If they're all part of the sample gradle project, you should be able to just do a
```
./gradlew appengineDeploy
```
from the root and it'll run the task on all 3 projects.

The ear project is no longer necessary in this case.

Will add documentation soon : https://github.com/GoogleCloudPlatform/app-gradle-plugin/issues/87

**Comment by swftvsn** [comment]
Hi, 

that's great news! Thanks for the swift responses!

TL;DR: I got it working and am eagerly waiting for the documentation to add some finesse to my build process, as now it contains copied sections here and there.

Here are some corner cases I observed:

1. All projects (that are deployed to appengine) must have the app-gradle-plugin, or else the exploded directory is not created
1. All projects (that are deployed to appengine) must declare the same appYamls conf in run section, or else things break if the order of projects is changed. (Things work as long as the one having the correct conf is used to run, but as soon as it changes the other projects are not found. This can happen if the project that is used to run it is not explicitly set when invoking cradlew)

The first and also the second part may very well be me not knowing how to share confs as I'm coming from maven world. Anyhow, if I try to extract the configuration to super project, some error always occurs.

**Comment by patflynn** [comment]
@swftvsn thanks for these details.  We also have a goal to improve the UX of these plugins for multi-module configs this quarter.  We'll link in the issues once they're created.
